Modules/mathmodule.c

```diff
@@ -2231,14 +2226,12 @@ loghelper(PyObject* arg, double (*func)(double))
         }
 
         x = PyLong_AsDouble(arg);
-        if (x == -1.0 && PyErr_Occurred()) {
-            if (!PyErr_ExceptionMatches(PyExc_OverflowError))
-                return NULL;
+        if (PyErr_Occurred()) {
             /* Here the conversion to double overflowed, but it's possible
                to compute the log anyway.  Clear the exception and continue. */
             PyErr_Clear();
             x = _PyLong_Frexp((PyLongObject *)arg, &e);
-            if (x == -1.0 && PyErr_Occurred())
+            if (PyErr_Occurred())
```

Comment:
Same as above. `_PyLong_Frexp()` should be fixed if it returns not -1.0 on error.

Reply:
See comment on above change.